pkg/controller/custom.go

```diff
@@ -108,17 +108,21 @@ func (c *Controller) getCustomCertificates(ctx context.Context, tenant *miniov2.
 				expiresInSeconds := int64(math.Mod(expiresIn.Seconds(), 60))
 				expiresInHuman := fmt.Sprintf("%v days, %v hours, %v minutes, %v seconds", expiresInDays, expiresInHours, expiresInMinutes, expiresInSeconds)
 
-				if expiresInDays >= 10 && expiresInDays < 30 {
-					c.recorder.Event(tenant, corev1.EventTypeWarning, "CertificateExpiring", fmt.Sprintf("%s certificate '%s' is expiring in %d days", certType, secret.Name, expiresInDays))
-				}
-				if expiresInDays > 0 && expiresInDays < 10 {
-					c.recorder.Event(tenant, corev1.EventTypeWarning, "CertificateExpiryImminent", fmt.Sprintf("%s certificate '%s' is expiring in %d days", certType, secret.Name, expiresInDays))
+				if tenant.Spec.CertExpiryAlertThreshold == nil || expiresInDays < *tenant.Spec.CertExpiryAlertThreshold {
+					if expiresInDays >= 10 && expiresInDays < 30 {
+						c.recorder.Event(tenant, corev1.EventTypeWarning, "CertificateExpiring", fmt.Sprintf("%s certificate '%s' is expiring in %d days", certType, secret.Name, expiresInDays))
+					}
+					if expiresInDays > 0 && expiresInDays < 10 {
+						c.recorder.Event(tenant, corev1.EventTypeWarning, "CertificateExpiryImminent", fmt.Sprintf("%s certificate '%s' is expiring in %d days", certType, secret.Name, expiresInDays))
+					}
```

Comment:
Shouldn't these values be related to the `Spec.CertExpiryAlertThreshold` value?

Reply:
Some companies have rather short certificate expirations and rotate certificates often, so that's probably why this PR is needed. I thought we wanted to control when these messages occur, but I guess this is fine too. Maybe @jiuker is right and another name would make it more obvious.